lib/presto-client/index.js

```diff
@@ -113,7 +108,10 @@ Client.prototype.request = function(opts, callback) {
     if (opts.body)
         contentBody = opts.body;
 
-    opts.agent = new adapter.Agent(opts); // Otherwise SSL params are ignored.
+    opts.agents = {
+        http: new http.Agent({ keepAlive: false }),
+        https: new https.Agent({ ...client.ssl, keepAlive: false }),
```

Comment:
I decided to not pass in the `opts` object as is into the agent, as I don't think most users would need to have `host`, `port`, etc. set explicitly on the agent, vs letting that get picked up automatically from the request object. If wanted to do this, then would need to override the `port` value of `opts` if `opts.protocol === 'http:'`, which just seemed overly complicated for little benefit. Additionally, passing the `client.ssl` object into the `http.Agent` was probably a no-op in the common case as the values there were ignored, so I think better to pass it only into the `https.Agent`.

Note, when passing no agent, the default behavior was changed in Node 20 such that `keepAlive: true` is used, so I still pass in an agent to explicitly turn that off to maintain BC, as well as since these are all one off requests, there's no point to having `keepAlive: true`.

Finally, as a note, doing a spread of `undefined` returns empty object essentially, so if `client.ssl === undefined`, then doing `https.Agent({ ...client.ssl, keepAlive: false }) === https.Agent({ keepAlive: false })` essentially.

Reply:
This `opts.agents` attribute is obviously different from other opts attributes because others are to specify http/https request parameters, but this one is to control the `follow-redirects`-specific behaviors.
So I want a comment here about what this controls.

Reply:
I've added a comment in bcd7789c50a823291441821818b3b41ea63e433c, let me know if you'd like it tweaked.